### PR TITLE
chore: Commit application files during the connect flow

### LINKED ISSUE #12259: [Task]: Commit application files in connect flow to avoid errorneous state for discard changes
### Is there an existing issue for this?

- [X] I have searched the existing issues

### SubTasks

Today as we are not committing the application files and only the readme file in the connect flow, this has introduced a bug in the discard flow as we try to reconstruct the application from the last commit present on the remote which includes just a readme file.

### PR DESCRIPTION
**Pull request by abhvsn**
## Description

During the connect flow we are only committing the readme file and not the complete application JSON, as we wanted the user to have the first commit when the application is in a good shape. This brought 2 issues:
1. Users can connect 2 different applications to the same repo which is being tracked in https://github.com/appsmithorg/appsmith/issues/11615
2. We enable the discard changes CTA if the application has uncommitted changes. Now considering we only commit the readme file to verify the write access to the SSH key, we always get uncommitted changes because DB is not dehydrated yet.
To fix the above issues we are now committing the application resources along with the readme file in the connect flow itself.

Fixes #12259 #11615 #11266

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Update JUnit TC
> Manual test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Test coverage results :test_tube:
<details><summary>:white_circle: Total coverage has not changed</summary>

    // Code coverage diff between base branch:release and head branch: chore/commit-app-files-in-connect 
Status | File | % Stmts | % Branch | % Funcs | % Lines 
 -----|-----|---------|----------|---------|------ 
 :green_circle: | app/client/src/utils/WorkerUtil.ts | 89.76 **(0.78)** | 70.59 **(0)** | 100 **(0)** | 93.33 **(0.95)**</details>

### CODE REVIEW

**Review by AnaghHegde — COMMENTED** [review]
IIUC, we are using amend to commit the application files to the same commit. What will be the current time for the overall flow? I am just concerned what will happen if the app is very heavy, do we end up in timeout state?

**Comment by abhvsn** [comment]
Yes we are trying to use the git amend here, first commit only includes the readme file the second commit which is actually committing the application files takes same time as for commit and push. Today the git APIs are having the timeout of 2mins so we are good here even if 2nd commit and push takes more than 10sec which I observed for large apps like `community issues`

**Review by AnaghHegde — DISMISSED** [review]

**Comment by abhvsn** [comment]
@IAmAnubhavSaini can you please check if the testcase failures for git-sync are valid ones here? We are now committing the application files in the connect flow itself which results in `isClean: true` for the `getStatus API`

**Comment by IAmAnubhavSaini** [comment]
@abhvsn I am not seeing all files going in the first commit. I still have to commit manually from Deploy.

https://user-images.githubusercontent.com/1573771/161002237-941e4ffc-8694-4c71-b09e-7fde332813ab.mov

**Comment by abhvsn** [comment]
Are you pointing to release or running the server in your local?

**Comment by IAmAnubhavSaini** [comment]
@abhvsn one issue might be that this branch is way far behind from release. 

```
This branch is 2 commits ahead, 100 commits behind release.
```

You should rebase your changes on release, and then fire the UI testing again.

**Comment by IAmAnubhavSaini** [comment]
> Are you pointing to release or running the server in your local? I think you are still running on an older server build

Its the DP from this page.

**Review by AnaghHegde — APPROVED** [review]

**Comment by abhvsn** [comment]
@haojin111 did you get time to fix the issue we are facing, as we are seeing a test failure here.

**Comment by haojin111** [comment]
@abhvsn , yes, I will. I think it'll be resolved after following pr merged.
https://github.com/appsmithorg/appsmith/pull/12739
I fixed the cypress test issues on this pr.

**Comment by Parthvi12** [comment]
@haojin111  The test is still failing as "Latest Deploy Preview" option is not there on gitSync modal, and in the test we are trying to assert and open it.
Screenshot 2022-04-27 at 1 14 54 PM

File: app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/GitSync/GitBugs_spec.js [hunk 1]
```diff
@@ -166,7 +166,7 @@ describe("Git Bug: Fix clone page issue where JSObject are not showing up in des
   });
 });
 describe("Git synced app with JSObject", function() {
-  it("Create an app with JSObject, connect it to git and verify its data in edit and deploy mode", function() {
+  it.only("Create an app with JSObject, connect it to git and verify its data in edit and deploy mode", function() {
```
Comment (IAmAnubhavSaini) [inline app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/GitSync/GitBugs_spec.js]: `only` should be removed.

Comment (haojin111) [reply app/client/cypress/integration/Smoke_TestSuite/ClientSideTests/GitSync/GitBugs_spec.js]: already removed!

**Comment by IAmAnubhavSaini** [comment]
> @haojin111 The test is still failing as "Latest Deploy Preview" option is not there on gitSync modal, and in the test we are trying to assert and open it. Screenshot 2022-04-27 at 1 14 54 PM

The string "Changes since last commit" should not appear.

**Comment by haojin111** [comment]
passed unit test now
test_success

**Comment by IAmAnubhavSaini** [comment]
@AnaghHegde could you provide a build and deploy it to ec2 instance. I would like to test it.

**Comment by abhvsn** [comment]
> @AnaghHegde could you provide a build and deploy it to ec2 instance. I would like to test it.

@AnaghHegde have you deployed this or should I go ahead

**Review by IAmAnubhavSaini — APPROVED** [review]

**Pull request merged**